ci-operator/step-registry/openshift/image-based/upgrade/create-seed/openshift-image-based-upgrade-create-seed-commands.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -x
+set -o nounset
+set -o errexit
+set -o pipefail
+export PS4='+ $(date "+%T.%N") \011'
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-privatekey")
+
+PULL_SECRET_FILE="${CLUSTER_PROFILE_DIR}/pull-secret"
+PULL_SECRET=$(cat ${PULL_SECRET_FILE})
+BACKUP_SECRET_FILE="/var/run/ibu-backup-secret/.backup-secret"
+BACKUP_SECRET=$(cat ${BACKUP_SECRET_FILE})
+SEED_VM_NAME="seed"
+remote_workdir=$(cat ${SHARED_DIR}/remote_workdir)
+instance_ip=$(cat ${SHARED_DIR}/public_address)
+host=$(cat ${SHARED_DIR}/ssh_user)
+ssh_host_ip="$host@$instance_ip"
+
+seed_kubeconfig=${remote_workdir}/ib-orchestrate-vm/bip-orchestrate-vm/workdir-${SEED_VM_NAME}/auth/kubeconfig
+
+SEED_IMAGE_TAG="unknown"
+case $SEED_IMAGE_TAG_FORMAT in
+  "latest")
+    SEED_IMAGE_TAG="latest"
+    ;;
+  "nightly")
+    SEED_IMAGE_TAG="nightly-$(date +%F)"
+    ;;
+  "presubmit")
+    SEED_IMAGE_TAG="pre-${PROW_JOB_ID}"
+    ;;
+  *)
+    echo "Unknown image tag format specified ${SEED_IMAGE_TAG_FORMAT}"
+    exit 1
+    ;;
+esac
+
+echo "Creating seed script..."
+cat <<EOF > ${SHARED_DIR}/create_seed.sh
+#!/bin/bash
+set -euo pipefail
+
+# uncomment for debugging
+# set -x
+
+export PULL_SECRET='${PULL_SECRET}'
+export BACKUP_SECRET='${BACKUP_SECRET}'
+export SEED_VM_NAME="${SEED_VM_NAME}"
+export SEED_VERSION="${SEED_VERSION}"
+export LCA_IMAGE="${LCA_PULL_REF}"
+
+cd ${remote_workdir}/ib-orchestrate-vm
+
+# Create the seed vm
+make seed
+
+# Create and push the seed image
+make trigger-seed-image-create SEED_IMAGE=${SEED_IMAGE}:${SEED_IMAGE_TAG}
+
+echo "Waiting for seed creation to finish"
+until oc --kubeconfig ${seed_kubeconfig} wait --timeout 5m seedgenerator seedimage --for=condition=SeedGenCompleted=true; do \
```

Comment:
I'm not sure 5 minutes is long enough, also you might log "cluster unavailable" in case the 5 minutes timeout expires, in any case it seems that in case of failure to generate the seed the job will hang here until the job timesout (not a bad thing but I guess we can fail the job explicitly after some time e.g. 30 minutes).
Also, consider logging the CR status while we wait here, so in case it fails it will be easy to find what went wrong

Reply:
5 Minutes is definitely not enough time but for some reason when I wait longer than 5 minutes the script hangs (potentially a bug in the api server but I haven't had time to debug). I considered printing the CR status but typically this output triggers because the api server is down during seed creation so a status call to check the CR is going to fail anyway. In the event of a failure I think it makes sense to grab that data in a must-gather (which is a separate ticket that I will tackle once the presubmit jobs are in)